Tests/test_numpy.py

```diff
@@ -89,6 +115,24 @@ def test_16bit(self):
         self._test_img_equals_nparray(img, np_img)
         self.assertEqual(np_img.dtype, numpy.dtype('<u2'))
 
+    def test_save_tiff_uint16(self):
+        def _test_image(data, image):
+            '''
+            Check that saving a given image works assuming it's filled with 'data'
+            '''
+            tmpfile = self.tempfile("temp.tif")
+            a = numpy.array([data] * 100, dtype=numpy.uint16)
+            a.shape = TestNumpy.TEST_IMAGE_SIZE
+            Image.fromarray(a).save(tmpfile)
+            im_test = Image.open(tmpfile)
+            im_good = Image.open(image)
+            self.assert_image_equal(im_good, im_test)
+
+        _test_image(0, "Tests/images/uint16_1_0.tif")
+        _test_image(255, "Tests/images/uint16_1_255.tif")
+        _test_image(65280, "Tests/images/uint16_1_65280.tif")
+        _test_image(65535, "Tests/images/uint16_1_65535.tif")
```

Comment:
Let me add the comment that I'd like to keep the image the same repeated uint16 value tiled over a 10x10 image for consistency with other tests. I'll condense this to a single image, however.